File: src/main/java/com/huangyueran/spark/operator/Distinct.java
package com.huangyueran.spark.operator;

import com.huangyueran.spark.utils.SparkUtils;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.VoidFunction;

import java.util.Arrays;
import java.util.List;

/**
 * @category 对原RDD进行去重操作，返回RDD中没有重复的成员---Performs a reset operation on the original RDD and returns no duplicate members in the RDD
 * @author huangyueran
 * @time 2019-7-21 16:38:20
 */
public class Distinct {

	public static void main(String[] args) {
		JavaSparkContext sc = SparkUtils.getLocalSparkContext(Distinct.class);

		distinct(sc);
	}

	private static void distinct(JavaSparkContext sc) {
		List<String> datas = Arrays.asList("张三", "李四", "tom", "张三");

		 /**
		 *  ===================================
		 *   |      去重--包含shuffle操作                                                 |
		 *   |      Remove weights, including shuffle operations    |                                                                                                                                                                                                                                    | 
		 *   ===================================
		 */
		JavaRDD<String> distinctRDD = sc.parallelize(datas).distinct();
		
		distinctRDD.foreach(new VoidFunction<String>() {
			@Override
			public void call(String t) throws Exception {
				System.out.println(t);
			}
		});
	}

}

File: src/main/scala/com/hyr/spark/operator/Distinct.scala
package com.hyr.spark.operator

import com.hyr.spark.utils.SparkUtils
import org.apache.spark.{SparkConf, SparkContext}

/** *****************************************************************************
  * @date 2019-08-08 09:12
  * @author: <a href=mailto:huangyr>黄跃然</a>
  * @Description: 对原RDD进行去重操作，返回RDD中没有重复的成员---Performs a reset operation on the original RDD and returns no duplicate members in the RDD
  * *****************************************************************************/
object Distinct {

  def distinct(sparkContext: SparkContext): Unit = {
    val datas = List("张三", "李四", "tom", "张三")
    val rdd = sparkContext.parallelize(datas)
    val distinctRDD = rdd.distinct()
    distinctRDD.foreach(tuple => {
      println(tuple)
    })
  }

  def main(args: Array[String]): Unit = {
    val sparkContext = SparkUtils.getLocalSparkContext(Distinct.getClass)
    distinct(sparkContext)
  }

}
